### DISCUSSION TITLE
Runtime warning about destination state absence is not output

### DISCUSSION THREAD

**Discussion by takehilo** [pointfreeco/swift-composable-architecture#2896] [Q&A] [answered]
When I send an action to the store that has nil state, I get the following error on the Xcode console:

```
<decode: bad range for [%@] got [offs:394 len:833 within:0]>
```

The expected output is like the following:

```
An "ifLet" at "LoginCore/LoginCore.swift:79" received a presentation action when destination state was absent. …
...
```

You can reproduce this with the TicTacToe example.
Modify the following code, then run the app and tap "Log in".

```diff
- @ViewAction(for: Login.self)
public struct LoginView: View {
  // ...
  Button {
-    send(.loginButtonTapped)
+    store.send(.twoFactor(.presented(.view(.submitButtonTapped))))  // This is just an example.
```

This might be the cause.
https://github.com/pointfreeco/swift-composable-architecture/blob/main/Sources/ComposableArchitecture/Reducer/Reducers/PresentationReducer.swift#L592

**Comment by mbrandonw** [comment]
Hi @takehilo, I'm not sure there is much we can do about this. It seems like Xcode has changed its behavior. I'm curious if Apple's own purple runtime warnings have this behavior too. I don't have time to check, but do you?

**Reply by takehilo** [reply, marked as answer]
I tried this and fixed it!
https://stackoverflow.com/questions/78129981/logging-error-failed-to-initialize-logging-system-log-messages-may-be-missing
